### PR TITLE
[ENHANCEMENT] Use PBKDF2 for password hashing to make wallets slower to brute force

### PR DESCRIPTION
**Pull request by zpalmtree**
Currently using 500,000 iterations which seems relatively speedy to perform whilst making brute forcing a lot slower.

### CODE REVIEW

File: CantiLib/Blockchain/Crypto/JSONWalletEncrypter.cs [hunk 1]
```diff
@@ -46,6 +41,10 @@ namespace Canti.Blockchain.Crypto
 
     public class JSONWalletEncrypter : FileEncrypter<WalletBackend>
     {
+        /* The number of iterations to use when hashing the password with
+           PBKDF2 - intended to take roughly 1 second */
+        private const int PBKDF2Iterations = 500_000;
```
Comment (brandonlehmann) [inline CantiLib/Blockchain/Crypto/JSONWalletEncrypter.cs]: What's the `_` for?

Comment (zpalmtree) [reply CantiLib/Blockchain/Crypto/JSONWalletEncrypter.cs]: To make long numbers more readable.

1_000_000 == 1000000

Way easier to read how many zeros there are.

Comment (brandonlehmann) [reply CantiLib/Blockchain/Crypto/JSONWalletEncrypter.cs]: Is this a 🇬🇧 thing?

Comment (zpalmtree) [reply CantiLib/Blockchain/Crypto/JSONWalletEncrypter.cs]: No, it's a C# thing, lol

**Pull request merged**